src/interfaces/interfaces.ts

```diff
@@ -40,6 +40,8 @@ namespace interfaces {
 
   export type Newable<T> = new (...args: never[]) => T;
 
+  export type Instance<T> = T & Record<string, () => void>;
```

Comment:
Interesting choice here. I think there may be a better name for this type as it relates to the context it is used in.

1. I like the idea of encapsulating this type with `type` keyword.
2. Calling it `Instance` feels a little undercooked since the right hand side of the assignment does not really imply to me that this type is exlusively `Instance`-like. Does it seem possible that we could infer a better name from its usage?
3. Should we consider whether this should be exported publicly from the package? We tend to put all interfaces in this module, which makes them available to consumers of the package, but thinking of proper encapsulation, does a consumer of the package care about this type? Is there a better/closer place to put?

I'm personally ready to challenge the assumption that using `interfaces.ts` for all interfaces and types is the correct pattern, but it's certainly not in scope for this one issue :) Maybe we should keep it here for now and thinking about breaking apart `interfaces` in a future breaking changes release?